macro/src/lib.rs

```diff
@@ -114,10 +114,10 @@ impl Parse for BoundedInteger {
         } else {
             end
         };
-        if start >= end {
+        if start > end {
             return Err(Error::new_spanned(
                 range,
-                "The start of the range must be before the end",
+                "The start of the range must be before or equal to the end",
```

Comment:
It still can't be equal to if the range is exclusive.

Reply:
Personally, I would move this check before the inclusive/exclusive range adjustment and make it change the error based upon which kind of range is being used. That way it's clearest to the user.